lib/AST/Expr.cpp

```diff
@@ -1795,14 +1790,14 @@ ConditionalCheckedCastExpr *
 ConditionalCheckedCastExpr::create(ASTContext &ctx, SourceLoc asLoc,
                                    SourceLoc questionLoc, TypeRepr *tyRepr) {
   return new (ctx) ConditionalCheckedCastExpr(nullptr, asLoc, questionLoc,
-                                              new (ctx) TypeExpr(tyRepr));
+                                              tyRepr);
 }
 
 ConditionalCheckedCastExpr *
 ConditionalCheckedCastExpr::createImplicit(ASTContext &ctx, Expr *sub,
                                            Type castTy) {
   auto *const expr = new (ctx) ConditionalCheckedCastExpr(
-      sub, SourceLoc(), SourceLoc(), TypeExpr::createImplicit(castTy, ctx));
+      sub, SourceLoc(), SourceLoc(), nullptr);
```

Comment:
You need to set the cast type in here. One of the reasons I used TypeExpr was because it automatically maintained the invariant that implicit TypeExpr nodes were constructed with their types already in place. Forgetting this will break CSGen in spectacular ways now.